### ISSUE TITLE
Heroes screen and Unit charges on their icon

### ISSUE THREAD

**Issue by loks1234** [civfanatics/CQUI_Community-Edition#262]
**Describe the bug**
Heroes screen is super stretched out. Might be only for me because I'm using ultrawide screen(21:9). Builder and Religious Unit charges on their unit icon not working after the Feb 2021 update.

**To Reproduce**
1.Open the Great people window and click on Heroes Tab(Maybe only on ultrawide. Not sure on 16:9 screen)

**Expected behavior**
Heroes window is not supposed to stretch out but remain the same width as the original great people window.
Unit charges on the unit icon.

**General informations**
 - OS: Windows 10
 - Version 1.0.10.15
 - Expansion Gathering stoem
 - Sukritacts ui adjustments

**Save game and logs**

[BÀ TRIỆU 54 1880 BC.zip](https://github.com/civfanatics/CQUI_Community-Edition/files/6050358/BA.TRI.U.54.1880.BC.zip)

**Screenshots**
Great People window:
![Screenshot (29)](https://user-images.githubusercontent.com/14857152/109315070-235e8780-7870-11eb-939c-05435d990537.png)
Heroes window:
![Screenshot (28)](https://user-images.githubusercontent.com/14857152/109315110-2e191c80-7870-11eb-8608-e3d5146de5b3.png)

Builder charges info missing:
![Screenshot (30)](https://user-images.githubusercontent.com/14857152/109315157-3b360b80-7870-11eb-8c96-a14253fc75ea.png)

**Comment by loks1234** [comment]
Charges are displayed correctly after uninstalling and reinstalling mod from steam workshop

**Comment by the-m4a** [comment]
Hmm... yeah it's supposed to stay the same width, I think.  I'm not sure if I forced it to stretch the width on purpose, I'll look into it.

**Comment by the-m4a** [comment]
yeah, the width change is on purpose.  There's a calculation to determine the width to set that entire panel, and in order to be efficient with the code it makes that calculation based on the content inside that control.  Since the individual entries for the heroes are set wider, it means the panel holding all of those instances is willing to grow as wide as the screen would allow.

Changing that to always keep the same size as the initial panel... that may be doable, I think?  I just spent about 10 minutes trying to remember how to navigate that code - not so much that it's a mess, just there's so many controls in that panel it's hard to keep track of what to get the right size on the thing.

To be clear - it's not the setting of the panel width that's the pain, it's the getting the textures and just inside of it to correctly show is the pain... for example - I can just force the thing to be 600px wide, and I thought the various code I have in place prior to that _should_ paint those background textures correctly... but no, it does not:
![image](https://user-images.githubusercontent.com/8787640/110296799-95894780-7fa7-11eb-9b08-9410b23a81f0.png)

**Comment by the-m4a** [comment]
or actually... okay this is why it is confusing sometimes.  I cause the panel to refresh and now it's okay:
![image](https://user-images.githubusercontent.com/8787640/110297126-fadd3880-7fa7-11eb-9d2c-7806d3f91c7f.png)

Hmm...  I'll think about this one.  Basically it's a matter of keeping track of whatever size was calculated by the usual first panel, instead of recalculating that width each time the hero panel is shown.

**Comment by loks1234** [comment]
Awesome. Can you make the individual heroes have the same width as the individual great people ones then. Even though the heroes info is more than the great people, we don't need to view their info always since they stay the same unlike the individual great people info. Then even if the window is stretched it won't look as odd as it is now.

**Comment by the-m4a** [comment]
I like the wider heroes panel part - all that information shown just felt so squished when it used the same width as the Great People.    I do like the idea of at least keeping the panel the same width for each tab, it is jarring to have it change width when a different tab is selected.  Reverting the individual people sizes - or at least providing the option of "original" or "wide" size is maybe something that wouldn't be too bad to implement.  I'll think about it -- either way, I will hold off on any of these changes until I finish up with the barbarian tribes thing and the issue with the units that should be invisible not staying that way after applying the religion lens (that change is in the unitflagmanager file, which will also have changes for the barbarian thing and I'm hoping it's a small fix...)

**Comment by loks1234** [comment]
No hurry man, take your time.👍
